asciidocapi.py

```diff
@@ -177,28 +177,42 @@ def __init__(self, asciidoc_py=None):
         self.options = Options()
         self.attributes = {}
         self.messages = []
-        # Search for the asciidoc command file.
-        # Try ASCIIDOC_PY environment variable first.
+        # Search for the asciidoc command file in that order :
+        # - ASCIIDOC_PY environment variable
+        # - asciidoc_py function argument
+        # - sibling (preferred to shell search paths, to ensure version matching)
+        # - shell search paths
+        # - current working directory (may be different from the directory of this file)
+        fnames = ['asciidoc.py', 'asciidoc.pyc', 'asciidoc']
         cmd = os.environ.get('ASCIIDOC_PY')
+        is_cmd_valid = lambda c: c and os.path.isfile(c)
         if cmd:
             if not os.path.isfile(cmd):
                 raise AsciiDocError('missing ASCIIDOC_PY file: %s' % cmd)
-        elif asciidoc_py:
```

Comment:
whats wrong with the simple `elif`? better than defining a lambda function and then calling it.

Reply:
as mentioned in the commit message,  `else` are more error prone : https://www.maxpou.fr/else-unnecessary-expression
But, I agree, its a matter of taste.